EIPS/eip-7658.md

```diff
@@ -0,0 +1,183 @@
+    process_block_header(state, block)
+    ...
+```
+
+##### New `process_best_sync_data`
+
+```python
+def process_best_sync_data(state: BeaconState, block: BeaconBlock) -> None:
+    signature_period = compute_sync_committee_period_at_slot(block.slot)
+    attested_period = compute_sync_committee_period_at_slot(state.latest_block_header.slot)
+
+    # Track sync committee finality
+    old_has_sync_committee_finality = state.parent_block_has_sync_committee_finality
+    if state.parent_block_has_sync_committee_finality:
+        new_has_sync_committee_finality = True
+    elif state.finalized_checkpoint.epoch < ALTAIR_FORK_EPOCH:
+        new_has_sync_committee_finality = False
+    else:
+        finalized_period = compute_sync_committee_period(state.finalized_checkpoint.epoch)
+        new_has_sync_committee_finality = (finalized_period == attested_period)
+    state.parent_block_has_sync_committee_finality = new_has_sync_committee_finality
+
+    # Track best sync data
+    if attested_period == signature_period:
+        max_active_participants = len(block.body.sync_aggregate.sync_committee_bits)
+        new_num_active_participants = sum(block.body.sync_aggregate.sync_committee_bits)
+        old_num_active_participants = sum(state.current_best_sync_data.sync_aggregate.sync_committee_bits)
+        new_has_supermajority = new_num_active_participants * 3 >= max_active_participants * 2
+        old_has_supermajority = old_num_active_participants * 3 >= max_active_participants * 2
+        if new_has_supermajority != old_has_supermajority:
+            is_better_sync_data = new_has_supermajority
+        elif not new_has_supermajority and new_num_active_participants != old_num_active_participants:
+            is_better_sync_data = new_num_active_participants > old_num_active_participants
+        elif new_has_sync_committee_finality != old_has_sync_committee_finality:
+            is_better_sync_data = new_has_sync_committee_finality
+        else:
+            is_better_sync_data = new_num_active_participants > old_num_active_participants
+        if is_better_sync_data:
+            state.current_best_sync_data = SyncData(
+                sync_aggregate=block.body.sync_aggregate,
+                signature_slot=block.slot,
+            )
+```
+
+## Rationale
+
+### How to rank `SyncAggregate`?
+
+The EIP reuses the [`is_better_update`](https://github.com/ethereum/consensus-specs/blob/4afe39822c9ad9747e0f5635cca117c18441ec1b/specs/altair/light-client/sync-protocol.md#is_better_update) function from existing specs.
+
+### How could a backfill protocol use this?
+
+Once the data is available in the `BeaconState`, a light client data backfill protocol could be defined that serves, for past periods:
+
+1. A `LightClientUpdate` from requested `period` + 1 that proves that the entirety of `period` is finalized.
+2. `BeaconState.historical_summaries[period].block_summary_root` at (1)'s `attested_header.beacon.state_root` + Merkle proof.
```

Comment:
the summary period and light client periods are not same

Reply:
They are the same on all networks. The only LC period that's shorter is the initial one in case Altair got scheduled outside a sync committee period / historical summaries boundary.

Reply:
ahh yes, one is in slots and other in epochs,